run.sh

```diff
@@ -8,6 +8,7 @@ docker run -it --rm \
 --pid=host \
 --env UID=$(id -u) \
 --env GID=$(id -g) \
+--env ROS_DOMAIN_ID=22 \
```

Comment:
This will just change the ROS_DOMAIN_ID for the development container and not for all containers being launched.

I'd suggest to not merge this change to main, but rather add a variable
--env ROS_DOMAIN_ID to all run.sh files, which maps the env variable from the host to the containter and also add to the compose.yaml file a setting

```
environment:
  ROS_DOMAIN_ID
```


This way all containers will use the ROS_DOMAIN_ID variable as set on the host running the containers, no matter if you start dev containers with ./run.sh or via docker-compose.

Reply:
Yeah this was just to get it running temporarily on my system, it shoudn't have been added here sorry. The way you described it sounds good, i'll try to get that set up first before debugging the description & bringup further.

Reply:
However, it works is fine. I guess being able to set a custom ROSdomid on the host in one place is generally a useful thing to have, especially when you said other robots might interfere in your local domain with it.